plugins/org.jkiss.dbeaver.model.ai/src/org/jkiss/dbeaver/model/ai/engine/copilot/CopilotClient.java

```diff
@@ -221,6 +225,64 @@ public Flow.Publisher<CopilotChatChunk> createChatCompletionStream(
         return publisher;
     }
 
+    /**
+     * Retrieves a list of Copilot model IDs. If the current model list is empty or if a force refresh
+     * is required, it attempts to fetch and load models using the provided authentication token.
+     *
+     * @param monitor the progress monitor to report task progression and check for task cancellation
+     * @param token an authentication token used to fetch the models; can be null
+     * @param forceRefresh a boolean flag indicating whether to force a refresh of the model list
+     * @return a list of model IDs as strings; an empty list is returned if no models are available
+     */
+    public static List<String> getModels(@NotNull DBRProgressMonitor monitor, @Nullable String token, boolean forceRefresh) {
```

Comment:
Oh man, this method's signature gives me shivers. Let's see:
1. If we want to load models for the first time, we'd have to pass `monitor` and `token`
2. If we want to **re**load models, we'd have to pass `monitor`, `token` and `forceRefresh=true`
3. If we want to load cached models, we'd have to pass just the `monitor` that wouldn't be even used.

I see that models are cached in a static field. The issue is, depending on the token, a different set of models can be available. So, if you were to change the token, it _could_ theoretically yield completely different models. So it's not ideal to make it `static`, or cache it **at all**.